src/version.rs

```diff
@@ -7,7 +7,7 @@ use std::fmt;
 use self::HttpVersion::{Http09, Http10, Http11, Http20};
 
 /// Represents a version of the HTTP spec.
-#[derive(PartialEq, PartialOrd, Copy, Debug)]
+#[derive(PartialEq, PartialOrd, Copy, Clone, Debug)]
```

Comment:
Eq, Ord, and Hash too while you're at it? It seems valid to want to put these in a hashmap for routing of some kind.